Life refactor and slime as animal

phil235 opened the pull request:
Move most of carbon/Life() to living/Life().
Refactor robot/Life() and simple_animal/Life(). 
Makes all slimes into simple animals.
Fixes #8241. Fixes #8050. Fixes #8054
Reduces the number of vars used for simple_animal mobs.
Moving xenobiology items from slimes.dm to modules/research/xenobiology/xenobiology.dm
Refactor attack_slime, especially for carbons.
Removing simple_animal/Die(). That code is now handled in simple_animal/death().
Remove crit status to brain and slime mobs, reducing slime's overall health.

phil235 committed:
WIP moving carbon/life to living/life()

phil235 committed:
More work on living/life.dm(), simple_animal/life() and robot/life().

phil235 committed:
slime from carbon to simple animal

phil235 committed:
More work on slimes. Moving xenobiology items to modules/research/xenobiology

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SlimeAnimal

Conflicts:
	code/modules/mob/living/carbon/alien/life.dm
	code/modules/mob/living/carbon/life.dm
	code/modules/mob/living/carbon/slime/slime.dm
	code/modules/mob/living/silicon/robot/life.dm
	code/modules/mob/living/simple_animal/slime/life.dm

phil235 committed:
Fix conflict in xenobiology.dm
Simplify simple_animal/attackby()
more work on slimes.

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SlimeAnimal

phil235 committed:
More work on slime procs.

phil235 committed:
Yet more work on slime procs and life subprocs.

phil235 committed:
small adjustments

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SlimeAnimal

Conflicts:
	code/modules/mob/living/carbon/life.dm
	code/modules/mob/living/carbon/slime/slime.dm
	code/modules/mob/living/simple_animal/friendly/corgi.dm
	code/modules/mob/living/simple_animal/slime/say.dm
	code/modules/reagents/Chemistry-Reagents/Consumable-Reagents/Food-Reagents.dm

phil235 committed:
More fixes

phil235 committed:
fixing more stuff

JJRcop commented on code/modules/mob/living/simple_animal/simple_animal.dm:
```diff
@@ -35,21 +35,11 @@
-	var/max_co2 = 5
-	var/min_n2 = 0
-	var/max_n2 = 0
+	var/list/atmos_requirements = list("min_oxy" = 5, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0) //Leaving something at 0 means it's off - has no maximum
```
Did you mean to rebalance max_tox, or was it by mistake? (Used to be 1, now 0)

phil235 replied:
woops, yeah it's a mistake

JJRcop commented on code/modules/mob/living/simple_animal/friendly/drone/_drone.dm:
```diff
@@ -174,4 +172,7 @@
\ No newline at end of file
+	return 2
+
+/mob/living/simple_animal/drone/handle_temperature_damage()
```
This mean drone not burns?

JJRcop commented on code/modules/mob/living/simple_animal/simple_animal.dm:
```diff
@@ -35,21 +35,11 @@
 	//Temperature effect
 	var/minbodytemp = 250
 	var/maxbodytemp = 350
-	var/heat_damage_per_tick = 3	//amount of damage applied if animal's body temperature is higher than maxbodytemp
```
I think people would prefer these instead of hard coded proc rewrites that you can't modify at runtime

phil235 replied:
I don't think people actually edit those two vars on a live server very often, plus they can still modify minbodytemp and maxbodytemp to change a mob's reaction to different temperatures.

WJohn commented:
Holy shit.

MrPerson commented:
All those find+replace mob/living/carbon/slime -> mob/living/simple_animal/slime is why I always suggest people use the goddamn helper procs instead of doing the istype directly.

RemieRichards commented:
Absolutely based :+1:

Miauw62 commented:
>slimes as simple_animals


![](http://suptg.thisisnotatrueending.com/archive/24949706/images/1369119459877.jpg)

WJohn commented:
Now to wait for bots to be turned into simple animals.

palpatine213 commented:
@WJohn An unintended side effect of that would be making the sentience potion work on bots. Although a sentient beepsky would be funny as hell.

optimumtact commented:
Remove MOB 2014

phil235 committed:
Adding a changelog.
Some work on death() procs of many animals.
Replacing most istype(XYZ, mob/living/simple_animal/slime) by isslime(XYZ).
Some work on slime feeding procs.
Made simple animal aliens see in the dark like regular aliens (in case somone wants to control one).

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SlimeAnimal

Conflicts:
	code/modules/mob/living/carbon/human/human.dm
	code/modules/mob/living/carbon/slime/slime.dm
	code/modules/mob/living/simple_animal/slime/say.dm

phil235 committed:
Fixes more merge conflicts

phil235 committed:
Adds a temporary fix to the conflict with metastation map.
Small fix in simple_animal life subproc.

phil235 commented:
good to go

optimumtact commented:
@phil235 https://tgstation13.org/phpBB/viewtopic.php?f=10&t=3290&p=79464#p79429

Some feedback on slimes
